src/__tests__/__actions__/searchActions.test.js

```diff
@@ -0,0 +1,195 @@
+      payload: value
+    };
+    expect(actions.handleInputChange(value)).toEqual(expectedActions);
+  });
+
+  test("should dispatch ARTICLE_SEARCH_SUCCESS and clear existing one", () => {
+    const expectedActions = [
+      {
+        type: SEARCHING_ARTICLES
+      },
+      {
+        type: CLEAR_SEARCH_RESULTS
+      },
+      {
+        type: ARTICLE_SEARCH_SUCCESS,
+        payload: {
+          articles: { ...arrayToObject(articles, "id") },
+          authors: { ...authors }
+        }
+      }
+    ];
+    moxios.stubRequest(
+      `${process.env.API_BASE_URL}/articles?keyword=${keyword}&page=${1}`,
+      {
+        status: 200,
+        response: {
+          articles
+        }
+      }
+    );
+    store = mockStore({});
+    return store
+      .dispatch(actions.fetchResults(keyword, undefined, history))
+      .then(() => {
+        expect(store.getActions()).toEqual(expectedActions);
+        expect(history.push).toHaveBeenCalledWith(`/search?keyword=${keyword}`);
+      });
+  });
+
+  test("should dispatch ARTICLE_SEARCH_SUCCESS to fetch more articles", () => {
+    const expectedActions = [
+      {
+        type: SEARCHING_ARTICLES
+      },
+      {
+        type: ARTICLE_SEARCH_SUCCESS,
+        payload: {
+          articles: { ...arrayToObject(articles, "id") },
+          authors: { ...authors }
+        }
+      }
+    ];
+    moxios.stubRequest(
+      `${process.env.API_BASE_URL}/articles?keyword=${keyword}&page=${2}`,
+      {
+        status: 200,
+        response: {
+          articles
+        }
+      }
+    );
+    store = mockStore({});
+    return store.dispatch(actions.fetchResults(keyword, 2)).then(() => {
+      expect(store.getActions()).toEqual(expectedActions);
+      expect(history.push).not.toHaveBeenCalledWith(
+        `/search?keyword=${keyword}`
+      );
+    });
+  });
+
+  test("should dispatch ARTICLE_SEARCH_FAILED action creator", () => {
+    const pageNumber = 1;
+    const expectedActions = [
+      {
+        type: SEARCHING_ARTICLES
+      },
+      {
+        type: ARTICLE_SEARCH_FAILED,
+        payload: {
+          message: "Unauthorized"
+        }
+      }
+    ];
+    moxios.stubRequest(
+      `${process.env.API_BASE_URL}${actions.searchURL(keyword, pageNumber)}`,
+      {
+        status: 401,
+        response: {
+          message: "Unauthorized"
+        }
+      }
+    );
+    store = mockStore({});
+    return store
+      .dispatch(actions.fetchResults(keyword, pageNumber, history))
+      .then(() => {
+        expect(store.getActions()).toEqual(expectedActions);
+        expect(history.push.mock.calls.length).toBe(1);
+      });
+  });
+
+  test("should dispatch SET_SUGGESTED_ARTICLES", () => {
+    const query = "hello world";
+    const expectedActions = [
+      {
+        type: SEARCH_QUERY_CHANGE,
+        payload: query
+      },
+      {
+        type: SET_SUGGESTED_ARTICLES,
+        payload: { articles: { ...arrayToObject(articles, "id") } }
```

Comment:
@abayo-luc, if the `arrayToObject()` returns an object, there is no need to spread it in an object. I think that you should rather assign its return value directly to the `articles` unless you have some additional properties that you need to give your `articles` object.